spec/unit/frontlight_spec.lua

```diff
@@ -0,0 +1,185 @@
+describe("Frontlight function in PowerD", function()
+    local PowerD, device
+    setup(function()
+        require("commonrequire")
+
+        PowerD = require("device/generic/powerd"):new{
+            frontlight = 0,
+        }
+
+        param = {
+            fl_min = 1,
+            fl_max = 5,
+            device = {
+                hasFrontlight = function() return true end
+            },
+        }
+    end)
+
+    before_each(function()
+        frontlight = param.fl_min
+        stub(PowerD, "init")
+        stub(PowerD, "frontlightIntensityHW")
+        stub(PowerD, "setIntensityHW")
+        PowerD.setIntensityHW = function(self, intensity)
+            self.frontlight = intensity
+        end
+        spy.on(PowerD, "setIntensityHW")
+        spy.on(PowerD, "turnOnFrontlightHW")
+        spy.on(PowerD, "turnOffFrontlightHW")
+    end)
+
+    it("should read frontlight intensity during initialization", function()
+        PowerD.frontlightIntensityHW.returns(2)
+        local p = PowerD:new(param)
+        assert.are.equal(p:frontlightIntensity(), 2)
```

Comment:
Although it'll fail due to a mismatch regardless if something changes, the busted pattern is always `(expected, obj)`. Otherwise it'll misleadingly say something like:

> Passed in: `2`
> Expected: whatever's the outcome of `p:frontlightIntensity()`

This obviously also applies to all of the following tests but I won't bother pointing them out individually.